### DISCUSSION TITLE
jquery & screenshot support

### DISCUSSION THREAD

**Discussion by Wachizungu** [buffer/thug#322] [Q&A]
Hey,

I was wondering if thug is supposed to be able to handle the latest version of jquery or not, I've been trying to analyze a page (for testing purposes) that uses jquery 3.6.0 and has a $(window).on('load') function.

I'm using thug -u win7ie100 -U -F <url>     but I keep getting  '[handle_events] Event onload not properly handled'.

Secondly, are screenshots still supported? (I installed imgkit and some other things that seemed to be dependencies for it but still get an error)

Thanks if you're willing and have the time to reply <3

-----------------------------

TL;DR of discussion results:
Bootstrap v5 gives some issues currently, jQuery should be okay!
if you are running Thug on Linux and want to use the screenshot option, make sure to install xvfb or it won't work and give a somewhat obscure error message

**Comment by buffer** [comment]
Hi,
I have not experienced issues with jquery recently but this does not mean that there could be some issues with it. Can you share a sample that could help me reproduce the issue? Even a minimized version is ok.

As per the screenshots, the feature is still supported but I have observed that the last version of imgkit (1.2.2) breaks my tests. Can you please try installing imgkit 1.1.0 and re-run your tests? Please let me know

**Reply by Wachizungu** [reply]
Thanks!

Sure, the page I'm currently testing with is https://cloaking.jeroenpinoy.website/cloakingsite/alert . JQuery should be at https://cloaking.jeroenpinoy.website/static/js/jquery.min.js . I am not directly seeing errors with that profile (there is an error when using the default one) , just that '[handle_events] Event onload not properly handled'.

Ok, I'll check with imgkit 1.1.0 asap.

**Comment by Wachizungu** [comment]
Okay rolled back to imgkit 1.1.0. Installed xvfb after some checking/digging, now it looks okay!

**Reply by buffer** [reply]
Sorry for the late reply but it's a really busy period for me. Great to know it works for you too! I will take care of the jquery issue as soon as I can.

**Reply by buffer** [reply]
I think I spotted the issue. Which is not related to jQuery but to Bootstrap v5.x. Performed multiple tests using a basic page and different versions of Bootstrap v4.x experiencing no issues at all. Taking a look at Bootstrap 5.0.2 was a great experience because I spotted some additional details about DOM internals I was not aware of. I am going to introduce such changes into the codebase in the next release(s). At the same time this could be marked as a low priority issue considering that Bootstrap v5 officially dropped support for IE 10 and 11. Thanks again for the great report!

**Reply by Wachizungu** [reply]
Ha, okay :D good to know (& a bit surprising to me as well). Thanks for the quick check / feedback! I'll redo my tests in the coming weeks with bootstrap v4 then.
